packages/documentation/plugins/remark/gatsby-remark-mdx-mermaid/index.js

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable */
+const visit = require('unist-util-visit');
+
+module.exports = ({ markdownAST }, { language = 'mermaid' } = {}) => {
+  visit(markdownAST, 'code', node => {
+    const lang = (node.lang || '').toLowerCase();
+
+    if (lang === language) {
+      node.type = 'html';
+      node.value = '<div class="mermaid">{' + '`'+ node.value + '`'+ '}</div>';
```

Comment:
This opens up some possibilities- I wonder if we could use mermaid in node to generate the svg markup from this `node.value` and just inline it here. That would remove the need to run the client side library. 

I'm not sure exactly what's being created in this context: just markup or React code. We can generate React friendly svgs if needed (we did this on the old veteran id project- some of the xml directives had to be excluded but otherwise it worked fine).

Reply:
@rianfowler I actually tried to do that but the only way to make it work with node is to use the mermaidCLI. I have't tried to implement the CLI yet but if we were to try to do it we would need to generate a .mmd file and then convert that into a svg and then add that link to our html document. Mermaid is client side and you won't be able to use it unless you have the window object and also the SVG object. I even tried it with the mermaidAPI using jsdom but that didn't have everything we needed to make it work. It would bomb out when trying to generate the SVG. I would be open to trying the CLI but I think that would take a little time to implement.

Reply:
Thanks for the info- that's really frustrating that it doesn't just generate the markup and requires DOM stuff. 

this approach is good- if we want we can try the CLI later. I tried the MMD file approach but I think the trade off isn't worth it unless we have a lot of diagrams. Would require a bunch of file structure or naming conventions that could be annoying.